### PR TITLE
Database unit tests

### PR DESCRIPTION
**Pull request by pfistererm**
Implementation of test cases that test the writing to the database.

## Type of request

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring 
- [ ] Documentation or documentation changes

## Concept 
New test case that writes to an in memory test database and checks the database contents afterwards.
#### Description of the change
Added test case and requirec configuration files.
#### Description of current state
No tests for database writing code in LicenseScout.
#### Initiator
Matthias Pfisterer

#### How has this Tested?
Running the new test case

## Checklist:

- [x] All the commits are signed.
- [x] My code follows the code style of this project.
- [x] I agree with die CLA.
- [x] I have read the CONTRIBUTING docs.
- [x] I have added/updated necessary documentation (if appropriate).

### CODE REVIEW

**Comment by pfistererm** [comment]
About the code quality issues regarding SQL files: they contain DBMS specific statements that are not available in ANSI SQL and therefore cannot be replaced by ANSI SQL. For these cases, the check "Checks SQL against the ANSI syntax" does not make sense. We should configure an exception.

**Comment by davu-all** [comment]
Please fix the failing checks

File: org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/core/test/util/TestUtil.java [hunk 1]
```diff
@@ -15,15 +15,24 @@
...
 import org.aposin.licensescout.util.ILFLog;
 import org.aposin.licensescout.util.JavaUtilLog;
 import org.aposin.licensescout.util.NullLog;
 
+//import java.util.logging.Logger;
+//
+//import org.aposin.licensescout.configuration.RunParameters;
+//import org.aposin.licensescout.util.ILFLog;
+//import org.aposin.licensescout.util.JavaUtilLog;
+//import org.aposin.licensescout.util.NullLog;
```
Comment (davu-all) [inline org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/core/test/util/TestUtil.java]: Is there any reason to keep those commented imports?

File: org.aposin.licensescout.core/src/test/resources/database/dataset.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<dataset>
+    <builds/>
+    <deectedlicenses/>
```
Comment (davu-all) [inline org.aposin.licensescout.core/src/test/resources/database/dataset.xml]: Is this a typo?

**Comment by davu-all** [comment]
@pfistererm Please fix the conflicts and the few points noted in the review. Thanks

File: org.aposin.licensescout.core/src/test/resources/database/dataset.xml (hunk 2 above)
Comment (pfistererm) [reply org.aposin.licensescout.core/src/test/resources/database/dataset.xml]: Yes, fixed it.

File: org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/core/test/util/TestUtil.java (hunk 1 above)
Comment (pfistererm) [reply org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/core/test/util/TestUtil.java]: No - left over. I deleted them.

**Pull request merged**